src/inclusion_proof.rs

```diff
@@ -56,38 +60,31 @@ pub fn generate_inclusion_proof(
     Ok(inclusion_proof_vec)
 }
 
-// function: verify_inclusion_proof
-// inputs: flat_file_directory, block_range, inclusion_proof
-// outputs: result
+/// verifies an inclusion proof generate by [`generate_inclusion_proof`]
+///
+/// * `blocks`-  A [`Vec<Block>`]. The blocks included in the inclusion proof interval, set in `start_block` and `end_block` of [`generate_inclusion_proof`]
+/// * `master_accumulator_file`- An instance of [`MasterAccumulator`] which is a file that maintains a record of historical epoch
+/// it is used to verify canonical-ness of headers accumulated from the `blocks`
+/// * `inclusion_proof` -  The inclusion proof generated from [`generate_inclusion_proof`].
 pub fn verify_inclusion_proof(
-    directory: &String,
-    master_accumulator_file: Option<&String>,
-    start_block: usize,
-    end_block: usize,
+    blocks: Vec<Block>,
+    master_accumulator_file: Option<MasterAccumulator>,
     inclusion_proof: Vec<[H256; 15]>,
 ) -> Result<(), EraValidateError> {
-    let num_blocks = end_block - start_block;
-
-    // Load master accumulator
-    let master_accumulator = match master_accumulator_file {
-        Some(master_accumulator_file) => {
-            MasterAccumulator::try_from_file(master_accumulator_file.into())
-                .map_err(|_| EraValidateError::InvalidMasterAccumulatorFile)?
-        }
+    let master_acc = match master_accumulator_file {
+        Some(master_acc) => master_acc,
         None => MasterAccumulator::default(),
     };
 
-    // Verify inclusion proof
-    let blocks = extract_100_blocks(&directory, start_block, end_block)?;
-    for block_idx in 0..num_blocks {
+    for (block_idx, _) in blocks.iter().enumerate() {
         let bhp = BlockHeaderProof::AccumulatorProof(AccumulatorProof {
-            proof: inclusion_proof[block_idx].clone(),
+            proof: inclusion_proof[block_idx as usize].clone(),
         });
         let hwp = HeaderWithProof {
-            header: header_from_block(blocks[block_idx].clone())?,
+            header: header_from_block(&blocks[block_idx as usize].clone())?,
             proof: bhp,
         };
-        master_accumulator
+        master_acc
             .validate_header_with_proof(&hwp)
             .map_err(|_| EraValidateError::ProofGenerationFailure)?;
```

Comment:
Another error type might be informative here, e.g. `EraValidateError::ProofValidationFailure`